cmd/multiFaExtract/multiFaExtract.go

```diff
@@ -5,54 +5,102 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/common"
 	"github.com/vertgenlab/gonomics/fasta"
 	"log"
 )
 
-func multiFaExtract(infile string, outfile string, start int, end int, removeGaps bool) {
-	if !(start < end) {
-		log.Fatalf("Invalid arguments, start must be lower than end")
-	}
-	records := fasta.Read(infile)
+func multiFaExtract(s Settings) {
 	var ans []fasta.Fasta
-	for i := 0; i < len(records); i++ {
-		ans = append(ans, fasta.Extract(records[i], fasta.RefPosToAlnPos(records[0], start), fasta.RefPosToAlnPos(records[0], end), records[i].Name))
-	}
-	if removeGaps {
-		ans = fasta.RemoveGaps(ans)
+	records := fasta.Read(s.InFile)
+	if s.Bed == "" {
+		if !(s.Start < s.End) {
+			log.Fatalf("Invalid arguments, start must be lower than end")
+		}
+		ans = extractMultiHelper(records, s.Start, s.End)
+		if s.RemoveGaps {
+			ans = fasta.RemoveGaps(ans)
+		}
+		fasta.Write(s.OutFile, ans)
+	} else {
+		bedChan := bed.GoReadToChan(s.Bed)
+		for b := range bedChan {
+			ans = extractMultiHelper(records, b.ChromStart, b.ChromEnd)
+			if s.RemoveGaps {
+				ans = fasta.RemoveGaps(ans)
+			}
+			fasta.Write(fmt.Sprintf("%s.%v.%v.fa", b.Chrom, b.ChromStart, b.ChromEnd), ans)
+		}
 	}
+}
 
-	fasta.Write(outfile, ans)
+func extractMultiHelper(records []fasta.Fasta, start int, end int) []fasta.Fasta {
+	var ans = make([]fasta.Fasta, len(records))
+	for i := range records {
+		ans[i] = fasta.Extract(records[i], fasta.RefPosToAlnPos(records[0], start), fasta.RefPosToAlnPos(records[0], end), records[i].Name)
+	}
+	return ans
 }
 
 func usage() {
 	fmt.Print(
 		"multiFaExtract - Pull sub-sequence from multiple Fasta alignment for each entry. Uses reference indices, treating the first sequence as the reference.\n" +
 			"Usage:\n" +
 			"multiFaExtract multi.fa out.fa start end\n" +
+			"OR\n" +
+			"multiFaExtract -bed regions.bed multi.fa\n" +
 			"options:\n")
 	flag.PrintDefaults()
 }
 
+type Settings struct {
+	InFile     string
+	OutFile    string
+	Start      int
+	End        int
+	Bed        string
+	RemoveGaps bool
+}
+
 func main() {
 	var expectedNumArgs int = 4
 	var removeGaps *bool = flag.Bool("removeGaps", false, "Removes gaps from the output sequences. Note that the output will no longer be in valid multiFa format.")
+	var bed *string = flag.String("bed", "", "Extract all regions from start and end positions specified by the input bed file. All regions must be on the same chromosome as the multiFa (remember this is byChrom). Output files will be named according to the coordinates of the bed entry.")
```

Comment:
I think this is saying that the "chrom" field of the bed is ignored.  Maybe say that explicitly?

Reply:
Resolved in next commit